KioskApplication/src/main/java/controller/PathfindingSidebarController.java

```diff
@@ -135,4 +202,111 @@ public void resetScreen() {
+
+        class Wrapper<T> { T content ; }
+        Wrapper<MouseEvent> eventWrapper = new Wrapper<>();
+
+        PauseTransition holdTimer = new PauseTransition(holdTime);
+        holdTimer.setOnFinished(event -> handler.handle(eventWrapper.content));
+
+
+        node.addEventHandler(MouseEvent.MOUSE_PRESSED, event -> {
+            eventWrapper.content = event ;
+            holdTimer.playFromStart();
+        });
+        node.addEventHandler(MouseEvent.MOUSE_RELEASED, event -> holdTimer.stop());
+        node.addEventHandler(MouseEvent.DRAG_DETECTED, event -> holdTimer.stop());
+    }
+
+    /**
+     * create new waypoint list cell
+     */
+    private void newWaypointBox(Node node) {
+        HBox waypointBox = new HBox();
+
+        Label nodeNameLabel = new Label(node.getLongName());
+        nodeNameLabel.setTextFill(Color.BLACK);
+        nodeNameLabel.setStyle("-fx-font-weight:bold; "+
+                "-fx-font-size: 12pt; ");
+
+        JFXButton btRemoveWaypoint = new JFXButton("x");
+        btRemoveWaypoint.setStyle("-fx-background-color: #ff000e;"+
+                "-fx-font-size: 16pt; "+
+                "-fx-font-weight:bold; "+
+                "-fx-background-radius: 3em; ");
+//            btRemoveWaypoint.setGraphic(removeIconView);
+        btRemoveWaypoint.setPrefHeight(8);
+        btRemoveWaypoint.setPrefWidth(8);
+        btRemoveWaypoint.setTooltip(new Tooltip("Remove"));
+        btRemoveWaypoint.setOnAction(new EventHandler<ActionEvent>() {
+            @Override
+            public void handle(ActionEvent event) {
+                removeWaypoint(node);
+            }
+        });
+
+        waypointBox.getChildren().addAll(btRemoveWaypoint, nodeNameLabel);
+        waypointBox.setAccessibleText(node.getNodeID());
+        waypointBox.setMargin(btRemoveWaypoint, new Insets(1,1,1,1));
+        waypointBox.setMargin(nodeNameLabel, new Insets(10,1,1,10));
+        waypointListView.getItems().add(waypointBox);
+    }
+    /**
+     * remove the target waypoint bounded with input node
+     */
+    private void removeWaypoint(Node node) {
+        if(waypointListView.getItems().size()>=2) {
+            getMapController().removeWaypoint(currentNodes.get(currentNodes.size()-1));
+            waypointListView.getItems().remove(waypointListView.getItems().size()-2);
+            currentNodes.remove(currentNodes.size()-1);
+        }
+    }
+    /**
+     * create add waypoint list cell
+     */
+    //TODO make addwaypointbox always the last one
+    private void addWaypointBox() {
+//        int addWaypointBoxIndex = 0;
+        HBox addWaypointBox = new HBox();
+
+        Label addWaypointLabel = new Label("Click to Add a New Waypoint");
+        addWaypointLabel.setTextFill(Color.BLACK);
+        addWaypointLabel.setStyle("-fx-font-weight:bold; "+
+                "-fx-font-size: 12pt; ");
+        JFXButton btNewWayPoint = new JFXButton("+");
+        btNewWayPoint.setStyle("-fx-background-color: #00ff02;"+
```

Comment:
These should be moved into the CSS file and then the css class should be set somehow